### ISSUE TITLE
Move RPN and RCNN losses into `keras_rcnn.losses` module

### ISSUE THREAD

**Issue by 0x00b1** [broadinstitute/keras-rcnn#162]
The RPN and RCNN layers have methods for their respective losses (each has a classification_loss and a regression_loss). We should refactor the methods into functions and move them into the keras_rcnn.losses module to replicate the Keras interface.

They should also, wherever necessary, be refactored to conform to the conventional Keras loss API:

```Python
def name(y_true, y_pred):
	return …
```

In addition, I believe this is a useful starting point for completely replacing the RPN and RCNN layers with equivalent loss functions.
